Jetpack Agency Dashboard - Plugin Management: implement plugin management for multi-site view

yashwin opened the pull request:
#### Proposed Changes

This PR is built on top of https://github.com/Automattic/wp-calypso/pull/65720.

This PR implements plugin management UI from WP.com in the agency dashboard plugin management multi-site view

#### Testing Instructions

**Prerequisites**

Since this change is made specifically for agencies, you must set yourself(partner) as an agency - 2c49b-pb. Make sure to switch it back to the previous type.

**Instructions**

1. Run `git checkout add/import-plugin-management-for-agency-dashboard-multi-site-view` and `yarn start-jetpack-cloud`
2. Open http://jetpack.cloud.localhost:3000/, and you'll be redirected to the /dashboard.
3. Click on the **Plugins** nav item and verify the plugins are shown as shown below.

<img width="1920" alt="Screenshot 2022-07-19 at 4 59 00 PM" src="https://user-images.githubusercontent.com/10586875/179739789-408adebd-ebd4-4477-a8ac-38eb1ffd00c8.png">

4. Click on all the tabs - **All**, **Active**, **Inactive** & **Updates** and verify the plugins are shown as per the filter, and the URL is changing accordingly.
5. Click on the search icon, enter any search term and verify it is working as expected.
6. We have imported the component used here https://wordpress.com/plugins/manage, so there are not many UI changes except that we are hiding the top breadcrumbs.
7. Verify that the same is working as expected in Calypso Blue.
8. All the required style and UI enhancements will be done in the upcoming PRs

Mobile view

<img width="373" alt="Screenshot 2022-07-20 at 1 50 24 PM" src="https://user-images.githubusercontent.com/10586875/179934192-546f62b6-8d06-4140-8886-c6437d0c33d8.png">

<img width="379" alt="Screenshot 2022-07-20 at 1 50 31 PM" src="https://user-images.githubusercontent.com/10586875/179934217-d86d770d-9bcb-4b26-8e57-e610321001ac.png">


#### Pre-merge Checklist

Complete applicable items on this checklist **before** merging into the trunk. Inapplicable items can be left unchecked.

The PR author and reviewer are responsible for completing the checklist.

- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes? Will be added later - 1202518759611394-as-1202627702018877
- [x] ~~Have you tested the feature in [Simple](P9HQHe-k8-p2-p2), [Atomic](P9HQHe-jW-p2-p2), and [self-hosted Jetpack sites](PCYsg-g6b-p2-p2)?~~
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Are we memoizing when appropriate (for expensive computations)? More info in [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md) and [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors)
- [x] Have we sent any new strings [for translation](PCYsg-1vr-p2-p2) ASAP?

Related to 1202518759611394-as-1202534044814270

yashwin committed:
Add plugin management multi-site routes

yashwin committed:
Use in-built VoidFunction

rcanepa commented on client/my-sites/plugins/main.jsx:
```diff
@@ -139,28 +139,30 @@ export class PluginsMain extends Component {
+		const { translate, isJetpackCloud } = this.props;
 		const siteFilter = this.props.selectedSiteSlug ? '/' + this.props.selectedSiteSlug : '';
 
+		const basePath = isJetpackCloud ? '/dashboard/plugins' : '/plugins';
```
We can remove this since the path won't include `/dashboard` either.

rcanepa commented on client/my-sites/plugins/main.jsx:
```diff
@@ -431,15 +433,17 @@ export class PluginsMain extends Component {
-						{ this.renderUploadPluginButton( this.state.isMobile ) }
-					</div>
-				</FixedNavigationHeader>
+				{ ! this.props.isJetpackCloud && (
```
What do you think about changing the name of this prop from `isJetpackCloud` to `shouldDisplayNavigationHeader` (or something similar)?

If the only purpose of that prop is hiding/showing the navigation header (since now we don't need a different base path for the URL), a more specific name for the prop will result in a more generic approach.

By the way, I don't have a strong feeling about this.

yashwin replied:
I have made the changes and using `shouldDisplayNavigationHeader` in the child component so that it is easy to understand, but still keeping the props from the component as `isJetpackCloud` in the parent component since we will need it for other changes in the upcoming PRs

rcanepa commented:
On mobile, the UI has padding on the sides, which does not happen on WordPress.com. Is this expected?


<table>
<tr>
<th>
Jetpack cloud
</th>
<th>
WordPress.com
</th>
</tr>
<tr>
<td>
<img src="https://user-images.githubusercontent.com/3418513/179828445-d13984e8-a921-4583-b23a-29b33116ed55.png" width="500" />
</td>
<td>
<img src="https://user-images.githubusercontent.com/3418513/179828526-1765699e-c633-45f4-a0d2-25e281497bc6.png" width="500" />
</td>
</tr>
</table>

Besides this, everything is looking great.

yashwin committed:
Use /plugins as URL

Also, remove condition to check if plugin management is enabled since the component handles this scenario

yashwin committed:
Move plugin management into new module and add sections

yashwin commented:
> On mobile, the UI has padding on the sides, which does not happen on WordPress.com. Is this expected?

@rcanepa, as mentioned in the description, we will include all the UI enhancements in a different PR. This PR is mainly for implementing the importing of the component to Jetpack Cloud.

yashwin committed:
Implement existing plugin management UI from WP.com for multi-site view

yashwin committed:
Add sidebar navigation to the Plugins page

rcanepa reviewed (approved):
LGTM!
